fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/LogicalSortExpression.java

```diff
@@ -109,6 +119,22 @@ public Value getResultValue() {
         return inner.getFlowedObjectValue();
     }
 
+    @Override
+    public boolean semanticEquals(@Nullable final Object other, @Nonnull final AliasMap aliasMap) {
```

Comment:
So `semanticEquals()` in `RelationalExpression` deals with the entire DAG of expressions. You probably just want to override `equalsWithoutChildren()`

Reply:
Removed